### PR TITLE
xenclient-toolstack: Overhaul the recipe

### PR DESCRIPTION
**Pull request by eric-ch**
The only built components provided by xenclient-toolstack are ocaml libraries. The rest are scripts, most of which are either not installed or not deployed in the end.

The MACHINE specificity seems to only stand on an RDEPENDS requirement which is no ground for that sort of configuration, this can be handled directly by each image configuration, and an image specific script.

Introduce xenclient-ocaml-libs that will build the ocaml libraries that other projects depend upon. Since ocaml statically link, this should only end up in DEPENDS.

Introduce xenclient-<type>-scripts recipes that will move the scripts to the layer. xenclient-vif-scripts-{dom0,ndvm} takes care of the image specificity. the block and tap scripts are not used by blktap3, so not deployed in common installations, but are provided for people interested
in blktap2.

### CODE REVIEW

**Comment by eric-ch** [comment]
Build [$10057](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10057)

File: recipes-openxt/xenclient-vif-scripts/xenclient-vif-scripts-ndvm_1.0.bb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+inherit allarch
+
+do_install() {
+    install -m 0755 -d ${D}${sysconfdir}/udev
+    install -m 0755 -d ${D}${sysconfdir}/udev/rules.d
+    install -m 0644 ${S}/xen-vif-backend.rules ${D}${sysconfdir}/udev/rules.d/xen-vif-backend.rules
+
+    install -m 0755 -d ${D}${sysconfdir}/xen/scripts
+    install -m 0644 ${S}/vif ${D}${sysconfdir}/xen/scripts/vif
```
Comment (jandryuk) [inline recipes-openxt/xenclient-vif-scripts/xenclient-vif-scripts-ndvm_1.0.bb]: Need to install 0755

File: recipes-openxt/xenclient-vif-scripts/xenclient-vif-scripts-dom0_1.0.bb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+inherit allarch
+
+do_install() {
+    install -m 0755 -d ${D}${sysconfdir}/udev
+    install -m 0755 -d ${D}${sysconfdir}/udev/rules.d
+    install -m 0644 ${S}/xen-vif-backend.rules ${D}${sysconfdir}/udev/rules.d/xen-vif-backend.rules
+
+    install -m 0755 -d ${D}${sysconfdir}/xen/scripts
+    install -m 0644 ${S}/vif ${D}${sysconfdir}/xen/scripts/vif
```
Comment (jandryuk) [inline recipes-openxt/xenclient-vif-scripts/xenclient-vif-scripts-dom0_1.0.bb]: Need to install 0755

**Comment by eric-ch** [comment]
#### V2:
- Fixed permissions set on scripts from 644 to 755
- Renamed recipes to drop the `xenclient` naming.

**Comment by stacktrust** [comment]
OXT-1771 for the xenclient branding change

**Comment by jandryuk** [comment]
I was going to say "Will merge soon" after a local built.  However, there is now a mismatch between the commit message saying "xenclient-<type>-scripts" and the actual names "xen-<type>-scripts" after the recipe rename.

**Comment by eric-ch** [comment]
@jandryuk Damn, my bad, the commit message is now up to date.

**Comment by jandryuk** [comment]
No worries.  The force-push shows no code change.

Will merge soon.

**Pull request merged**